Assignment 3.Rmd

```diff
@@ -127,29 +133,213 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+person_person_matrix<-person_course%*%course_person
+
+#Creating a grap from an adjacency matrix(person-person)
+library(igraph)
+g<-graph_from_adjacency_matrix(person_person_matrix,mode="directed",diag=FALSE)
+plot(g,vertex.color=df1$fit2.cluster)
+```
 
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
 
+```{r}
+df_6<-read.csv("HUDK405019-clustering.csv",encoding ="UTF-8",stringsAsFactors = FALSE)
+df_6<-df_6%>%
+  unite(Full.Name,First.Name,Last.Name)
+
+#Convert the index numbers of the data frame into the student names.
+
+row.names(df_6) <- df_6$Full.Name
+
+df_6$Full.Name <- NULL
+
+#Wrangle data using dplyr to include only the numerical values.
+
+#Remove location variables
+df_7 <- dplyr::select(df_6, 1:11)
+
+#Remove any characters
+df_8<-lapply(df_7,function(x)gsub("[A-z]","",x))
+df_8<-data.frame(df_8)
+
+#Convert all variables to numeric
+for (i in 1:dim(df_8)[2]){
+  df_8[,i]<-as.numeric(as.character(df_8[,i])) 
+  #changing from factor;need to convert it into character first
+}
+
+#Scale the data so that no variable has undue influence
+df_8 <- data.frame(scale(df_8))
+ 
+#Replace missing values with average score EG - zero
+df_8 <- df_8 %>% mutate_all(funs(ifelse(is.na(.) == TRUE, 0, .)))
+
+#Handling columns with latitues and longtitudes
+df_9 <- dplyr::select(df_6, 13:14)
+
+#Change names for convenience
+names(df_9) <- c("lattitude", "longitude")
+
+#Remove any characters and common punctuation
+df_9 <- df_9 %>% mutate_all(funs(gsub("[a-zA-Z]", "", .)))
+df_9 <- df_9 %>% mutate_all(funs(sub("[?]", "", .)))
+
+#Remove anything after the first non-numeric character in lattitude
+df_9$lattitude <- sub(",.*$","", df_9$lattitude) 
+df_9$lattitude <- sub("°.*$","", df_9$lattitude)
+
+#Remove anything before the first non-numeric character in longitude
+df_9$longitude <- gsub(".*,","",df_9$longitude)
+df_9$longitude <- sub("°.*$","", df_9$longitude)
+
+#Convert all variables to numeric
+df_9 <- df_9 %>% mutate_all(funs(as.numeric(.)))
+
+
+fit3<-kmeans(df_8,4)
+
+fit3$cluster
+
+df_10<-data.frame(df_8,df_9,fit3$cluster)
+
+#graph created in activity 6
+library(ggplot2)
+ggplot(df_10, aes(longitude, lattitude, color = as.factor(fit3.cluster))) +
+  geom_point(size = 3)
+
+#combining two cluster together
+Full.Name<-toupper(row.names(df_6))
+df_10<-data.frame(Full.Name,df_10)
+df_11<-inner_join(df_10,df1,by="Full.Name")
+df_11$Full.Name<-as.factor(1:36)
+
+
+#overlaying two ggplots
+#diamond:cluster 1 based on questionare
+#filled circle:cluster2 based on social network
+ggplot(df_11,aes(x=Full.Name,y=fit3.cluster,color=as.factor(fit3.cluster)))+
+  geom_point(size=5,pch=5)+
+  geom_point(data=df_11,aes(x=Full.Name,y=fit2.cluster,color=as.factor(fit2.cluster)),size=5,pch=19)+
+  xlab("Students")+
+  ylab("Clusters")+
+  labs(color='Diamond:Cluster1;Circle:Cluster2')
```

Comment:
Awesome work! You can also use the mosaic plot to visualize the overlap of clusters.